src/providers/ipa/ipa_views.c

```diff
@@ -183,21 +184,30 @@ static errno_t get_be_acct_req_for_xyz(TALLOC_CTX *mem_ctx, const char *val,
         break;
     default:
         DEBUG(SSSDBG_CRIT_FAILURE, "Unsupported request type [%d].\n", type);
-        talloc_free(ar);
-        return EINVAL;
+        ret = EINVAL;
+        goto done;
+    }
+
+    ret = sss_filter_sanitize(ar, val, &ar->filter_value);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "sss_filter_sanitize failed.\n");
+        goto done;
     }
 
-    ar->filter_value = talloc_strdup(ar, val);
     ar->domain = talloc_strdup(ar, domain_name);
     if (ar->filter_value == NULL || ar->domain == NULL) {
         DEBUG(SSSDBG_OP_FAILURE, "talloc_strdup failed.\n");
-        talloc_free(ar);
-        return ENOMEM;
+        ret = ENOMEM;
+        goto done;
     }
 
-
     *_ar = ar;
```

Comment:
I know that ret will be EOK here, but I'd still explicitly set ret = EOK (it makes the code clearer, IMO).